homeassistant/components/songpal/config_flow.py

```diff
@@ -63,12 +64,18 @@ async def async_step_user(self, user_input=None):
                         vol.Required(
                             CONF_ENDPOINT, default=user_input.get(CONF_ENDPOINT, "")
                         ): str,
+                        vol.Optional(
+                            CONF_URL, default=user_input.get(CONF_URL, "")
```

Comment:
Why do we need to ask the user for an URL and UUID? What do they represent?

Reply:
The grouping features require it, instead of the 'endpoint', it calls the upnp-url of the to-be 'master' and passes the to-be paired devices UUIDs as arguments

Reply:
And how would the user know about these things? At least the UUID I would expect to be retrieved from the device via the device library.

Reply:
The auto discover function should detect both, so only if you manually add you'd need to 'find' them (which is hard, yes, but can be done using discovery features of the underlying lib https://github.com/rytilahti/python-songpal)

Reply:
I don't think we should ask the user for this info. The integration should get this info from the device library, maybe by using the discovery interface.

Reply:
Clear, agree to some extent, as the current endpoint you need to fill is equally hard to find (e.g. `http://192.168.1.102:10000/sony` , but both port and 'sony' part differ per model and can't be found anywhere in user interface, only with discovery or the lib above). So, in general manual adding is hard/impossible for non-advanced users.